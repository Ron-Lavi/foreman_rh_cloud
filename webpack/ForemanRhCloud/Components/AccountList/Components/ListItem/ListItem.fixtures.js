export const props = { accountID: '1' };
