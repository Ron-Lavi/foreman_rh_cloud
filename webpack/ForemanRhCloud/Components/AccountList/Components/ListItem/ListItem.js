import React from 'react';
import { ListView } from 'patternfly-react';
import PropTypes from 'prop-types';
import ListItemStatus from '../ListItemStatus';
import Dashboard from '../../../Dashboard';

const ListItem = ({ accountID, account, initExpanded }) => (
  <ListView.Item
    leftContent={<ListView.Icon name="user" />}
    heading={account.label}
    additionalInfo={[
      <ListItemStatus key={`${accountID}_status`} account={account} />,
    ]}
    stacked
    hideCloseIcon
    initExpanded={initExpanded}
  >
    <Dashboard accountID={accountID} account={account} />
  </ListView.Item>
);

ListItem.propTypes = {
  accountID: PropTypes.string.isRequired,
  account: PropTypes.shape({
    generate_report_status: PropTypes.string,
    upload_report_status: PropTypes.string,
    label: PropTypes.string,
  }),
  initExpanded: PropTypes.bool,
};

ListItem.defaultProps = {
  account: {
    generate_report_status: 'unknown',
    upload_report_status: 'unknown',
    label: 'default_org_name',
  },
  initExpanded: false,
};

export default ListItem;
