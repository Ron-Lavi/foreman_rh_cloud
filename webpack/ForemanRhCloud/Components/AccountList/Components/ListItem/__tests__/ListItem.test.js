import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import ListItem from '../ListItem';
import { props } from '../ListItem.fixtures';

const fixtures = {
  'render with Props': props,
};

describe('ListItem', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(ListItem, fixtures));
});
