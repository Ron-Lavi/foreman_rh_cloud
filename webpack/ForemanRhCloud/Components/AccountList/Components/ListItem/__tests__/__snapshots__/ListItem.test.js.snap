// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem rendering render with Props 1`] = `
<ListViewItem
  actions={null}
  additionalInfo={
    Array [
      <ListItemStatus
        account={
          Object {
            "generate_report_status": "unknown",
            "label": "default_org_name",
            "upload_report_status": "unknown",
          }
        }
      />,
    ]
  }
  checkboxInput={null}
  compoundExpand={false}
  compoundExpanded={false}
  description={null}
  heading="default_org_name"
  hideCloseIcon={true}
  initExpanded={false}
  leftContent={
    <ListViewIcon
      className=""
      name="user"
      size="sm"
      type="fa"
    />
  }
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={true}
>
  <Connect(Dashboard)
    account={
      Object {
        "generate_report_status": "unknown",
        "label": "default_org_name",
        "upload_report_status": "unknown",
      }
    }
    accountID="1"
  />
</ListViewItem>
`;
